drivers/sdhc/sdhci_common.h

```diff
@@ -225,6 +227,29 @@
 #define SDHCI_SPEC_300         2
 #define SDHCI_SPEC_400         3
 
+/* ADMA2 descriptor */
+#define SDHC_ADMA_DESC_VALID      BIT(0)
+#define SDHC_ADMA_DESC_END        BIT(1)
+#define SDHC_ADMA_DESC_INT        BIT(2)
+#define SDHC_ADMA_DESC_ACT1       BIT(4)
+#define SDHC_ADMA_DESC_ACT2       BIT(5)
+#define SDHC_ADMA_DESC_ACT_NOP    0
+#define SDHC_ADMA_DESC_ACT_RSV    SDHC_ADMA_DESC_ACT1
+#define SDHC_ADMA_DESC_ACT_TRAN   SDHC_ADMA_DESC_ACT2
+#define SDHC_ADMA_DESC_ACT_LINK   SDHC_ADMA_DESC_ACT2 | SDHC_ADMA_DESC_ACT1
+#define SDHC_ADMA_DESC_LEN_SHIFT  16
+#define SDHC_ADMA_DESC_LEN_MASK   GENMASK(31, 16)
+#define SDHC_ADMA_DESC_ADDR_SHIFT 32
+#define SDHC_ADMA_DESC_ADDR_MASK  GENMASK(63, 32)
+
+#define SDHC_ADMA_DESC_MAX_LEN 65532
```

Comment:
> It's done as in u-boot. 

So, if you see someone jumping into a well, does that mean you should jump in after them?..

> To encode 65536 bytes, it should be handled as a special case by writing 0 in the length field. To avoid unnecessary checks, the descriptor data size is fixed to 65532 bytes.

This isn't true:
You have a 16-bit `len` field in an ADMA2 descriptor structure. What will happen if you write 65536 to it? What will actually be stored in the `len` field?

Reply:
The value is a little confusing to me without any explanation in the code as to why it was chosen. It's not a good idea to hardcode a strange value for everyone.

> So, zero length may be defined by the standard, but not all hardware supports it. The way it's currently implemented here is the most correct and safest.

I propose creating a configuration option so people can choose. You can define your own values for RPI5 and specify the limits in the configuration option. By default, choose something safe, for example, 32K, since you mentioned that this value is used by the Linux kernel.